### PR TITLE
454 bal datepicker buttons auto submit form

### LINKED ISSUE #454: [bal-datepicker] buttons auto-submit form
## Detailed description

The bal-buttons for the days in bal-datepicker have no type set and thus default to submit, creating submit-events on click.

## Steps to reproduce the issue

1. Use bal-datepicker in any form and listen for the submit event
2. Pick a date
3. Submit triggered on day button click

### PR DESCRIPTION
**Pull request by hirsch88**
Closes #

{{short description}}

## Changelog

**New**

- {{new thing}}

**Changed**

- {{change thing}}

**Removed**

- {{removed thing}}

## Acceptance Criteria

- [x] Design Review by @Gagne87, @clastzoo
- [x] Technical Review by @hirsch88, @mladenplaninicic, @nobilo
- Browser Testing by @hirsch88, @nobilo, @mladenplaninicic, @Gagne87, @clastzoo
    - Desktop
        - [x] Chrome
        - [x] Edge
        - [x] Safari
        - [x] Firefox
    - Tablet
        - [x] iPad (Landscape / Portrait)
    - Mobile
        - [x] Safari iOS
        - [x] Chrome Preview

### CODE REVIEW

File: packages/components/src/components/form/bal-checkbox/bal-checkbox-group/bal-checkbox-group.sass [hunk 1]
```diff
@@ -1,26 +1,26 @@
-+block(checkbox-group)
-  +unselectable
-  display: inline-block
-  position: relative
-  width: 100%
+// +block(checkbox-group)
```
Comment (mladenplaninicic) [inline packages/components/src/components/form/bal-checkbox/bal-checkbox-group/bal-checkbox-group.sass]: can we remove this commented style if we don't need it?

File: packages/components/src/components/form/bal-checkbox/bal-checkbox.sass [hunk 2]
```diff
@@ -70,85 +77,125 @@
...
-    border-top: 0
-    border-left: 0
-    -moz-transform: rotate(45deg)
-    -ms-transform: rotate(45deg)
-    -o-transform: rotate(45deg)
-    -webkit-transform: rotate(45deg)
-    transform: rotate(45deg)
-
-  input[type='checkbox']:not(.bal-checkbox__input--is-disabled) + label:hover
+  // &__label::before,
```
Comment (mladenplaninicic) [inline packages/components/src/components/form/bal-checkbox/bal-checkbox.sass]: can we remove this commented style if we don't need it?

File: packages/components/src/components/form/bal-radio/bal-radio copy.sass [hunk 3]
```diff
@@ -0,0 +1,225 @@
+@import './bal-radio.mixins'
+
```
Comment (mladenplaninicic) [inline packages/components/src/components/form/bal-radio/bal-radio copy.sass]: can we rename this sass file (just to remove 'copy' from it)?

File: packages/components/src/components/form/bal-radio/bal-radio copy.sass [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+//
+//
+// radio & checkbox
++block(radio-checkbox)
+  position: relative
+  display: inline-flex
+  justify-content: flex-start
+  align-items: center
+  outline: none
+  min-width: 24px
```
Comment (mladenplaninicic) [inline packages/components/src/components/form/bal-radio/bal-radio copy.sass]: can we use rem here?

File: packages/components/src/components/form/bal-radio/bal-radio.sass [hunk 5]
```diff
@@ -74,110 +81,256 @@
...
+  &--switch
+    .bal-radio-checkbox__label
+      padding-left: 1rem
+    .bal-radio-checkbox__label::before
+      border-radius: 24px
+      position: absolute
+      display: block
+      top: 0
+      left: 0
+      width: 40px
```
Comment (mladenplaninicic) [inline packages/components/src/components/form/bal-radio/bal-radio.sass]: can we use rem here too?

**Review by mladenplaninicic — APPROVED** [review]

File: packages/components/src/components/form/bal-radio/bal-radio.sass (hunk 5 above)
Comment (hirsch88) [reply packages/components/src/components/form/bal-radio/bal-radio.sass]: done

**Comment by hirsch88** [comment]
date picker will be fixed with another branch => https://github.com/baloise-incubator/design-system/issues/487

**Pull request merged**